### PR TITLE
Test re-parsing of all test files after optimization (with blacklist)

### PR DESCRIPTION
**Pull request by georgwiese**
A new iteration of #1476.

These added tests demonstrate that #1488 is *mostly* solved (see [this comment](https://github.com/powdr-labs/powdr/issues/1488#issuecomment-2220759508) though): We parse, optimize, serialize, and re-parse all test files. A small number of test files need to be blacklisted for reasons explained in the comments.

### CODE REVIEW

File: pipeline/tests/asm.rs [hunk 1]
```diff
@@ -527,6 +527,25 @@ fn vm_args_two_levels() {
     gen_estark_proof(f, Default::default());
 }
 
+mod reparse {
+
+    use powdr_pipeline::test_util::run_reparse_test_with_blacklist;
+    use test_log::test;
+
+    /// Files that we don't expect to parse, analyze, and optimize without error.
```
Comment (leonardoalt) [inline pipeline/tests/asm.rs:535]: Why not?

File: pipeline/tests/powdr_std.rs [hunk 2]
```diff
@@ -381,3 +381,26 @@ fn btree() {
...
+mod reparse {
+
+    use powdr_pipeline::test_util::run_reparse_test_with_blacklist;
+    use test_log::test;
+
+    /// For convenience, all re-parsing tests run with the Goldilocks field,
+    /// but these tests panic if the field is too small. This is *probably*
+    /// fine, because all of these tests have a similar variant that does
+    /// run on Goldilocks.
+    const BLACKLIST: [&str; 5] = [
```
Comment (leonardoalt) [inline pipeline/tests/powdr_std.rs:394]: Why these?

File: pipeline/tests/asm.rs (hunk 1 above)
Comment (georgwiese) [reply pipeline/tests/asm.rs:535]: The test failure cases

File: pipeline/tests/powdr_std.rs (hunk 2 above)
Comment (georgwiese) [reply pipeline/tests/powdr_std.rs:394]: They test machines designed to run on BN, so an assertion fails when run on GL.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**